cmd/nerdctl/save.go

```diff
@@ -103,16 +105,33 @@ func saveImage(images []string, out io.Writer, saveOpts []archive.ExportOpt, cmd
 	}
 
 	saveOpts = append(saveOpts, archive.WithPlatform(platMC))
-
 	imageStore := client.ImageService()
+	foundID := make(map[string]struct{})
+	walker := &imagewalker.ImageWalker{
+		Client: client,
+		OnFound: func(ctx context.Context, found imagewalker.Found) error {
+			imgName := found.Image.Name
+			imgDigest := found.Image.Target.Digest.String()
+			if _, ok := foundID[imgDigest]; !ok {
+				foundID[imgDigest] = struct{}{}
+			}
+			saveOpts = append(saveOpts, archive.WithImage(imageStore, imgName))
+			return nil
+		},
+	}
+
 	for _, img := range images {
-		named, err := referenceutil.ParseAny(img)
+		count, err := walker.Walk(ctx, img)
 		if err != nil {
 			return err
 		}
-		saveOpts = append(saveOpts, archive.WithImage(imageStore, named.String()))
+		if count == 0 {
+			return fmt.Errorf("no such image: %s", img)
+		}
+	}
+	if len(foundID) > len(images) {
+		return fmt.Errorf("ambiguous digest id")
```

Comment:
not sure but I think this is useless

Reply:
```
[root@k8s nmx]# nerdctl  -n k8s.io images|grep bc
calico/node                           v3.19.1                                                             bc4a631d553b    28 hours ago    linux/amd64    160.6 MiB    56.0 MiB
calico/pod2daemon-flexvol             v3.19.1                                                             bcac7dc4f130    28 hours ago    linux/amd64    20.7 MiB     8.9 MiB
sha256                                5660150975fb82f006d013fdb583b628d3d340ff64775ca3111ca28f523e32df    bcac7dc4f130    28 hours ago    linux/amd64    20.7 MiB     8.9 MiB
sha256                                c4d75af7e098eef5142c65b8fab5b3ff0e5c614ce72e91b8c44ba20249b596ef    bc4a631d553b    28 hours ago    linux/amd64    160.6 MiB    56.0 MiB
```
for example nerdctl -n k8s.io save  bc   -o test.tar  ,I want to show error.if not use this it wll not show error

Reply:
Checking `found.MatchCount > 1` might be enough

Reply:
ping